### DISCUSSION TITLE
Packaged .deb versions are not built with Yubikey support

### DISCUSSION THREAD

**Discussion by DDzwiedziu** [smallstep/certificates#2087] [General]
I was following the Tiny CA tutorial [1], but decided to take take pre-build .deb packages, with a slight anticipation that the upgrade could be done trough a repo, in some future.

Unfortunately ~3/4 into the tutorial starting up step-ca I've encountered this:

`unsupported kms type 'yubikey': step-ca is compiled without cgo or YubiKey support`

Now my question is is this a bug or a feature?
As I've expect a .deb package to be built against the maximum amount of features, to avoid such issues.

Edit: also I don't see the Go version mentioned in the build docs [2], which leaves me in the dark.

[1] https://smallstep.com/blog/build-a-tiny-ca-with-raspberry-pi-yubikey/
[2] https://github.com/smallstep/certificates/blob/master/CONTRIBUTING.md

**Comment by hslatman** [comment]
Hey @DDzwiedziu,

It was a deliberate choice to not have `yubikey` in the packages by default because of the additional maintenance time. Yubikey is not the only thing that requires additional maintenance, so things can get quite hairy supporting all possible combinations, and maintaining packages for all.

Generally our applications and libraries follow the Golang's standard practice of supporting the latest two (minor) versions.
